mailet/api/src/main/java/org/apache/mailet/AttributeValue.java

```diff
@@ -40,10 +42,13 @@
 
 /** 
  * Strong typing for attribute value, which represents the value of an attribute stored in a mail.
+ *
+ * Note that this class is Java serializable for compatibility purposes. Use at your own risks and always prefer relying on the JSON representation.
  * 
  * @since Mailet API v3.2
  */
-public class AttributeValue<T> {
+@Immutable
+public class AttributeValue<T> implements Serializable {
```

Comment:
There's 28 pages to read from Effective Java before you consider seriously this solution. There's so many pitfalls that I can't recall them all.